### PR TITLE
Locks the orientation if needed

### PR DESCRIPTION
**Pull request by Haitec**

### CODE REVIEW

**Comment by onmyway133** [comment]
@Haitec Hi, thanks for the PR. Could you tell the motivation behind this?

**Comment by Haitec** [comment]
In the project where I'm using it, our users are taking photos of railway stations.
We only use photos in landscape format.
So I integrated the functionality. :)

File: Source/CameraView/CameraMan.swift [hunk 1]
```diff
@@ -19,6 +20,10 @@ class CameraMan {
   var stillImageOutput: AVCaptureStillImageOutput?
   var startOnFrontCamera: Bool = false
 
+  init(withConfiguration configuration: Configuration) {
```
Comment (JohnSundell) [inline Source/CameraView/CameraMan.swift]: The Swift convention is to just use variable names as parameter labels in initializers. So here the `withConfiguration` can be dropped.

File: Source/Configuration.swift [hunk 2]
```diff
@@ -44,6 +44,7 @@ public struct Configuration {
   public var collapseCollectionViewWhileShot = true
   public var recordLocation = true
   public var allowMultiplePhotoSelection = true
+  public var lockedOrientation = UIInterfaceOrientationMask.all
```
Comment (JohnSundell) [inline Source/Configuration.swift]: Since this is a mask, how about naming it `allowedOrientations` instead?

File: Source/Helper.swift [hunk 3]
```diff
@@ -16,13 +18,89 @@ struct Helper {
...
-  static func videoOrientation() -> AVCaptureVideoOrientation {
-    switch UIDevice.current.orientation {
-    case .portrait: return .portrait
-    case .landscapeLeft: return .landscapeRight
-    case .landscapeRight: return .landscapeLeft
-    case .portraitUpsideDown: return .portraitUpsideDown
-    default: return .portrait
+  static func rotationTransform(_ configuration: Configuration) -> CGAffineTransform {
+    let currentOrientation = UIDevice.current.orientation
+    var result: CGAffineTransform
```
Comment (JohnSundell) [inline Source/Helper.swift]: Instead of this, you can imply return from each `case`

File: Source/Helper.swift [hunk 4]
```diff
@@ -16,13 +18,89 @@ struct Helper {
...
+
+    switch configuration.lockedOrientation {
```
Comment (JohnSundell) [inline Source/Helper.swift]: Couldn't this be simplified by simply checking if the current orientation is included in the allowed orientations mask?

File: Source/Helper.swift [hunk 5]
```diff
@@ -16,13 +18,89 @@ struct Helper {
...
+    case .landscapeRight:
+      return .landscapeLeft
+    case .portraitUpsideDown:
+      return .portraitUpsideDown
+    default:
+      return .portrait
+    }
+  }
+
+  static func videoOrientation(_ configuration: Configuration) -> AVCaptureVideoOrientation {
```
Comment (JohnSundell) [inline Source/Helper.swift]: How about moving this method to an extension on `Configuration`?

**Comment by Haitec** [comment]
I changed a few things. Maybe it's better now?

**Comment by zenangst** [comment]
Hey @Haitec, seems like we have some conflicts here again... mind fixing that up and then @JohnSundell can have a second look :)

**Comment by Haitec** [comment]
Updated. Time to shine for @JohnSundell :)

**Pull request merged**